[Capture] User transforms are expanded before execution (#7199)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h4>Capturing and Representing Hybrid Programs 📥</h4>

> * Applying transforms to a `QNode` now correctly transforms it with program capture enabled, rather than raising a
>   `NotImplementedError`.

Reviewer: 🤔 
```suggestion
* Transformations can now be directly applied to a `QNode` with program capture enabled.
```

Author: How about adding "without the use of the `@qml.capture.expand_plxpr_transforms` decorator"?
